From b24353da63d84420b5a103a91d475b4669ad4122 Mon Sep 17 00:00:00 2001
From: Zhike Wang <wangzk320@163.com>
Date: Fri, 8 Nov 2019 17:02:44 +0800
Subject: flow: Fix IPv6 header parser with partial offloading.

Set nw_proto before it is used in parse_ipv6_ext_hdrs__().

Signed-off-by: Zhike Wang <wangzk320@163.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 lib/flow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/flow.c b/lib/flow.c
index b7f35e33b..6c7c08fd8 100644
--- a/lib/flow.c
+++ b/lib/flow.c
@@ -1139,11 +1139,11 @@ parse_tcp_flags(struct dp_packet *packet)
         dp_packet_set_l2_pad_size(packet, size - plen);
         size = plen;
         const struct ovs_16aligned_ip6_frag *frag_hdr;
+        nw_proto = nh->ip6_nxt;
         if (!parse_ipv6_ext_hdrs__(&data, &size, &nw_proto, &nw_frag,
             &frag_hdr)) {
             return 0;
         }
-        nw_proto = nh->ip6_nxt;
     } else {
         return 0;
     }
-- 
2.14.1


